src/dsp_tools/utils/create_logger.py

```diff
@@ -41,5 +37,35 @@ def get_logger(
         backupCount=backupcount,
     )
     handler.setFormatter(formatter)
-    _logger.addHandler(handler)
-    return _logger
+
+    global rotating_file_handler
+    rotating_file_handler = handler
+
+
+def get_logger(
+    name: str,
+    level: int = logging.DEBUG,
+) -> logging.Logger:
+    """
+    Create a logger instance,
+    and configure it to write to a file in the user's home directory.
+
+    Args:
+        name: name of the logger
+        level: logging level, defaults to logging.DEBUG
+
+    Returns:
+        the logger instance
+    """
+    global rotating_file_handler
+    if not rotating_file_handler:
+        _make_handler(
+            Path.home() / Path(".dsp-tools"),
+            filesize_mb=20,
+            backupcount=10,
+        )
+        rotating_file_handler = cast(logging.handlers.RotatingFileHandler, rotating_file_handler)
```

Comment:
if you let the function return the handler instead of writing it to the global variable, and here you do `rotating_file_handler = _make_handler(...)`, then no cast is needed